__tests__/ncgamesapp.test.js

```diff
@@ -205,3 +204,49 @@ describe("GET:/api/reviews", () => {
       });
   });
 });
+describe("GET:/api/reviews/:review_id/comments", () => {
+  it("GET /api/reviews/:review_id/comments responds with correct number of comments for the specified review", () => {
+    return request(app)
+      .get("/api/reviews/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.comments).toHaveLength(3);
+      });
+  });
+  it("GET /api/reviews/:review_id/comments responds with correct categories", () => {
+    return request(app)
+      .get("/api/reviews/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        body.comments.forEach((comment) => {
+          expect(comment).toHaveProperty("author");
+          expect(comment).toHaveProperty("review_id");
+        });
+      });
+  });
+  it("GET /api/reviews/badid/comments string review_id throws 400 error", () => {
+    const newVote = { inc_votes: 1 };
+    return request(app)
+      .get("/api/reviews/badid/comments")
+      .expect(400)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("please enter valid review id");
+      });
+  });
+  it("GET /api/reviews/455/comments non-existing review_id throws 404 error", () => {
+    return request(app)
+      .get("/api/reviews/455/comments")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("review id not found");
+      });
+  });
+  it("GET /api/reviews/1/comments  review_id with no comments throws 404 error", () => {
+    return request(app)
+      .get("/api/reviews/1/comments")
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("no comments found");
```

Comment:
When we have a valid review with no comments on it we want to receive a 200 and an empty array. Imagine this being on a website would you expect an error if no one hasn't commented yet?